### PR TITLE
feat: DIA-1360: Add token cost KPI to the Prompt aggregate-level subset metrics

### PR DESCRIPTION
**Pull request by niklub**
| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit 88b07d851abef99438b1263fa8347bd81f4b7f5a  | 
|--------|--------|

### Summary:
This PR adds token cost tracking, updates runtime functions, modifies tests, and streamlines test setup.

**Key points**:
- Adds token cost tracking to Prompt aggregate-level subset metrics.
- Updates runtime functions in `adala/runtimes/_litellm.py`.
- Implements `_update_with_usage` for token usage and cost.
- Modifies `record_to_record` and `batch_to_batch` in `LiteLLMChatRuntime` and `AsyncLiteLLMChatRuntime`.
- Renames `process_file` to `stream_inference` in `docker-compose.yml`.
- Updates `server/handlers/result_handlers.py` for token cost fields in `LSEBatchItem`.
- Adds tests in `tests/test_llm.py` and `tests/test_stream_inference.py`.
- Streamlines test setup in `tests/test_server.py`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/601958b798cfe5d7987bcefcebc8f9b03dad5b5b is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/ddf571809b1f0e520a69af6caa659942f7a6774e is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/c459329e79e05a7104cc276fb3aea756f0dd3189 is created.

File: adala/runtimes/_litellm.py [hunk 1]
```diff
@@ -59,6 +84,35 @@ def _format_error_dict(e: Exception) -> dict:
...
 
+def _log_llm_exception(e) -> dict:
+    dct = _format_error_dict(e)
+    base_error = f"Inference error {dct['_adala_message']}"
+    tb = traceback.format_exc()
+    logger.error(f"{base_error}\nTraceback:\n{tb}")
+    return dct
+
+
+def _update_with_usage(data: Dict, usage: Usage, model: str) -> None:
```
Comment (pakelley) [inline adala/runtimes/_litellm.py:95]: nit: can we just return the usage dict and do something like `data.update(_update_with_usage(usage, model)`? I figure that's more transparent (likely would change the name of the function at that point, I guess)

File: adala/runtimes/_litellm.py [hunk 2]
```diff
@@ -33,6 +39,25 @@
 logger = logging.getLogger(__name__)
 
 
+# basically only retrying on timeout, incomplete output, or rate limit
+# https://docs.litellm.ai/docs/exception_mapping#custom-mapping-list
+# NOTE: token usage is only correctly calculated if we only use instructor retries, not litellm retries
+# https://github.com/jxnl/instructor/pull/763
+retry_policy = dict(
```
Comment (pakelley) [inline adala/runtimes/_litellm.py:46]: nit: can we make this in all caps to make it clear it's top-level scope? (I know we haven't been consistent about this in general, but I was looking around for it as a param and I think it'd help readability a bit)

File: server/tasks/stream_inference.py [hunk 3]
```diff
@@ -43,6 +44,29 @@ def parent_job_error_handler(self, exc, task_id, args, kwargs, einfo):
     delete_topic(output_topic_name)
 
 
+async def run_streaming(
```
Comment (pakelley) [inline server/tasks/stream_inference.py:47]: thanks for breaking this out 🙏

**Review by pakelley — APPROVED** [review]
couple minor things, but looks great overall

**Review by hakan458 — APPROVED** [review]

**Comment by luarmr** [comment]
@ellipsis-dev review this

**Review by matt-bernstein — APPROVED** [review]

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/0bbf95b73871f8b9206f4a129f4f7bd127edfbf8 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/38c566cfb367b0963bd7bc2435b140e4078a5cc1 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/64f4496d4bb4ebee76ac5c68e046e92c7f5447c9 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/b5beffe6dfee54e00961886f7de2f86af45f0026 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/09632d86e314b986d09f042e0328231431b32856 is created.

**Comment by robot-ci-heartex** [comment]
Follow Merge downstream workflow has been failed.
> [Workflow run](https://github.com/HumanSignal/label-studio-enterprise/actions/runs/10799741486)

**Pull request merged**